contracts/PeeranhaToken.sol

```diff
@@ -65,42 +92,93 @@ contract PeeranhaToken is ERC20Upgradeable, ERC20PausableUpgradeable, ERC20Cappe
+    } else if (boostContainer.updateBoostInPeriod.length == 0) {
+      boostContainer.updateBoostInPeriod.push(nextPeriod);
+    }
+
+    if (userBoost.updateUserBoostInPeriod.length != 0) {
+      uint16 lastBoostPeriod = userBoost.updateUserBoostInPeriod[userBoost.updateUserBoostInPeriod.length - 1];
+      if (lastBoostPeriod != nextPeriod) {
+        userBoost.updateUserBoostInPeriod.push(nextPeriod);
+      }
+
+      if (userBoost.stakedTokens[lastBoostPeriod] > stakeTokens) {
+        countBoost.allStake += userBoost.stakedTokens[lastBoostPeriod] - stakeTokens;
+        if (stakeTokens == 0) countBoost.countStaking--;
+      } else {
+        countBoost.allStake += stakeTokens - userBoost.stakedTokens[lastBoostPeriod];
+        if (userBoost.stakedTokens[lastBoostPeriod] == 0) countBoost.countStaking++;
+      }
+      userBoost.stakedTokens[nextPeriod] = stakeTokens;
+
+    } else {
+      userBoost.updateUserBoostInPeriod.push(nextPeriod);
+      userBoost.stakedTokens[nextPeriod] = stakeTokens;
+
+      countBoost.allStake += stakeTokens;
+      countBoost.countStaking++;
+    }
   }
 
   function getPool(RewardLib.WeekReward memory weekReward, uint16 period) private view returns(uint256) {
     uint256 poolToken = poolTokens[period];
     if (poolToken == 0) {
-      poolToken = reduceRewards(REWARD_WEEK, period);
+      poolToken = reduceRewards(REWARD_WEEK * FRACTION, period);                // max_pool
       if (weekReward.activeUsersInPeriod.length <= ACTIVE_USERS_IN_PERIOD) {
-        uint256 userRewardWeek = weekReward.activeUsersInPeriod.length * 1000 * FRACTION;
+        uint256 userRewardWeek = weekReward.activeUsersInPeriod.length * USER_REWARD * FRACTION;  // < 1000
         poolToken = CommonLib.minUint256(poolToken, userRewardWeek);
       }
-      poolToken = CommonLib.minUint256(poolToken, uint256(weekReward.rating) * TokenLib.getRewardCoefficient());
+      poolToken = CommonLib.minUint256(poolToken, weekReward.tokens);   // earned 
     }
 
     return poolToken;
   }
 
+  function reduceRewards(uint256 rewardWeek, uint16 period) private view returns(uint256) {
+    uint16 countReduce = period / 52;
+
+    for (uint16 i = 0; i < countReduce; i++) {
+      rewardWeek = (rewardWeek * 93) / 100;
+    }
+
+    return rewardWeek;
+  }
+
   function getUserReward(RewardLib.WeekReward memory weekReward, address user, uint16 period, uint256 poolToken) private view returns(uint256) {
     int32 ratingToReward;
-    uint256 tokenReward;
+    int32 tokenReward;
     uint32[] memory rewardCommunities = peeranha.getUserRewardCommunities(user, period);
-    for (uint32 i; i < rewardCommunities.length; i++) {   //вынести
+    for (uint32 i; i < rewardCommunities.length; i++) {
       ratingToReward = peeranha.getRatingToReward(user, period, rewardCommunities[i]);
-      if (ratingToReward == 0) continue;
-      tokenReward += uint256(ratingToReward);
+      tokenReward += ratingToReward;
```

Comment:
Can `ratingToReward` be negative here?
We should ignore negative values here I think.